lib/ansible/modules/cloud/vmware/vmware_vm_host_drs_rule.py

```diff
@@ -0,0 +1,467 @@
+    required: False
+    default: present
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Create DRS Affinity Rule for VM-Host
+  vmware_vm_host_drs_rule:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster_name: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    esxi_hostnames:
+        - DC0_H0
+    drs_rule_name: vm1-vm2-host1-affinity-rule-001
+    enabled: True
+    mandatory: True
+    affinity_rule: True
+
+- name: Create DRS Anti-Affinity Rule for VM-Host
+  vmware_vm_host_drs_rule:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster_name: "{{ cluster_name }}"
+    validate_certs: no
+    vms:
+        - vm1
+        - vm2
+    esxi_hostnames:
+        - DC0_H0
+    drs_rule_name: vm1-vm2-host1-anti-affinity-rule-002
+    enabled: True
+    mandatory: True
+    affinity_rule: False
+
+- name: Delete DRS Affinity Rule for VM-VM
+  vmware_vm_host_drs_rule:
+    hostname: "{{ esxi }}"
+    username: "{{ esxi_username }}"
+    password: "{{ esxi_password }}"
+    cluster_name: "{{ cluster_name }}"
+    validate_certs: no
+    drs_rule_name: vm1-vm2-host1-affinity-rule-001
+    state: absent
+'''
+
+RETURN = r'''#'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+from ansible.module_utils.vmware import (PyVmomi, vmware_argument_spec, wait_for_task,
+                                         find_vm_by_id, find_cluster_by_name)
+
+
+class VmwareDrs(PyVmomi):
+    def __init__(self, module):
+        super(VmwareDrs, self).__init__(module)
+        self.vm_list = module.params['vms']
+        self.hosts_list = module.params['esxi_hostnames']
+        self.cluster_name = module.params['cluster_name']
+        self.rule_name = module.params['drs_rule_name']
+        self.enabled = module.params['enabled']
+        self.mandatory = module.params['mandatory']
+        self.affinity_rule = module.params['affinity_rule']
+        self.state = module.params['state']
+
+        # Sanity check for cluster
+        self.cluster_obj = find_cluster_by_name(content=self.content,
+                                                cluster_name=self.cluster_name)
+        if self.cluster_obj is None:
+            self.module.fail_json(msg="Failed to find the cluster %s" % self.cluster_name)
+        # Sanity check for virtual machines
+        self.vm_obj_list = []
+        self.hosts_obj_list = []
+        if self.state == 'present':
+            # Get list of VMs only if state is present
+            # Create VM Group
+            self.vm_obj_list = self.get_all_vms_info()
+            self.vm_group_name = self.params.get('vm_group_name', None)
+            if self.vm_group_name is None:
```

Comment:
Can be covered in the `AnsibleModule` instantiation by using `required_if`

Reply:
Done.